### DISCUSSION TITLE
Replace flit with a packaging solution that supports editable installs

### DISCUSSION THREAD

**Discussion by ssbarnea** [fastapi/typer#566] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
-
```

### Description

Currently typer cannot be installed in editable mode because flit does not support dit

```
pip install -e .
ERROR: Project file:///Users/ssbarnea/c/os/typer has a 'pyproject.toml' and its build backend is missing the 'build_editable' hook. Since it does not have a 'setup.py' nor a 'setup.cfg', it cannot be installed in editable mode. Consider using a build backend that supports PEP 660.
```

I am suggesting setuptools-scm as that is easy to fix and its maintenance status is considerably better than the other ones I know.

If @tiangolo has nothing against, I would be more than happy to make a PR to do it as the switch should be quite easy.

### Operating System

Linux

### Typer Version

main branch

### Python Version

n/a

**Comment by ssbarnea** [comment]
To rephrase this issue, basically current typer packaging is discouraging contributions because it does not allow people to perform editable installs, which are the most effective way to test(experiment) changes made to a python library.

**Comment by connorferster** [comment]
I use `flit install --pth-file` for editable installs on projects I am working on with flit.

**Comment by svlandeg** [comment]
I'm also using `flit` and it's creating an editable install just fine on my system. @ssbarnea: did you follow the installation [instructions](https://typer.tiangolo.com/contributing/#flit)? Instead of the usual `pip install -e .`, running the `flit` command that is given there should work (note that the command looks slightly different on Windows vs Linux)

**Comment by tiangolo** [comment, marked as answer]
With the recent refactor from Flit to PDM, it is now installable with `pip`. You can do `pip install -e .`, or `pip install -r requirements.txt` to include the editable install and the development dependencies. :nerd_face: :rocket:
